aips/aip-x-safe-auth-key-mapping.md

```diff
@@ -0,0 +1,221 @@
+---
+aip: (this is determined by the AIP Manager, leave it empty when drafting)
+title: Safe onchain key rotation address mapping for standard accounts
+author: Alex Kahn (43892045+alnoki@users.noreply.github.com)
+Status: Draft
+type: Standard (Framework)
+created: 2024-09-17
+---
+
+# Problem statement
+
+Aptos authentication key rotation is accompanied by a global mapping from an
+authentication key to the address that it authenticates, the
+`OriginatingAddress` table. For more background see the [key rotation docs] and
+the [Ledger key rotation docs].
+
+There are currently several issues with the `OriginatingAddress` table (which is
+supposed to be a one-to-one lookup table) that render the mapping unsafe in
+practice:
+
+1. Per [`aptos-core` #13517], `rotate_authentication_key_call` does not update
+   the `OriginatingAddress` table for an "unproven" key rotation without a
+   `RotationProofChallenge` (resolved in this AIP's reference implementation
+   with a new `set_originating_address` private entry function).
+1. During an operation that attempts to map an authentication key (which has
```

Comment:
Upon a first read, and a subsequent one, I am not sure I understand this.

Are you saying something goes wrong when someone rotates two different accounts `a1` and `a2` to the same authentication key `ak`? Because the `OriginatingAddress` table `t` will be updated from `t[ak] = a1` to `t[ak] = a2`, breaking the (efficient) `a1` account recovery flow?

Reply:
(I think at least you are giving this example in the 3rd bullet point below?)

Reply:
And in the 2nd bulletpoint you are trying to argue that `OriginatingAddress` should not be overwritten?

Reply:
If so, I think there should be a much more concise way to explain this.

Reply:
Addressed in 4e5fb6a